conanfile_base.py

```diff
@@ -0,0 +1,87 @@
+import os
+import glob
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+
+
+class ConanfileBase(ConanFile):
+    _base_name = "flex"
+    name = "flex"
+    version = "2.6.4"
+    url = "https://github.com/bincrafters/conan-flex"
+    homepage = "https://github.com/westes/flex"
+    description = "Flex, the fast lexical analyzer generator"
+    topics = ("conan", "flex", "lex", "lexer", "lexical analyzer generator")
+    license = "BSD-2-Clause"
+    author = "Bincrafters <bincrafters@gmail.com>"
+    exports = ["LICENSE.md"]
+    exports_sources = ["patches/*.patch"]
+
+    _source_subfolder = "source_subfolder"
+
+    def source(self):
+        sha256 = "e87aae032bf07c26f85ac0ed3250998c37621d95f8bd748b31f15b33c45ee995"
+        tools.get("{0}/releases/download/v{1}/{2}-{3}.tar.gz".format(self.homepage, self.version, self._base_name, self.version), sha256=sha256)
+        extracted_dir = self._base_name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+
+    def requirements(self):
+        if tools.os_info.is_windows:
+            self.requires("pcre2/10.32@bincrafters/stable")
+
+    def _apply_patches(self):
+        for filename in sorted(glob.glob("patches/*.patch")):
+            self.output.info('applying patch "%s"' % filename)
+            tools.patch(base_path=self._source_subfolder, patch_file=filename)
+
+    @property
+    def cross_building(self):
+        if tools.cross_building(self.settings):
+            if self._the_os == self.detected_os():
+                if self._the_arch == "x86" and tools.detected_architecture() == "x86_64":
```

Comment:
`self._the_arch` wasn't declared before.

Reply:
fixed, thanks for pointing